### PR TITLE
issue #91 Improve class UserId

### PR DESCRIPTION
**Pull request by IvanPizhenko**

### CODE REVIEW

File: pgpainless-core/src/main/java/org/pgpainless/key/util/UserId.java [hunk 1]
```diff
@@ -13,113 +13,177 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package org.pgpainless.key.util;
 
+// @SuppressWarnings("unused")
```
Comment (vanitasvitae) [inline pgpainless-core/src/main/java/org/pgpainless/key/util/UserId.java]: This line can be deleted

File: pgpainless-core/src/main/java/org/pgpainless/key/util/UserId.java [hunk 2]
```diff
@@ -13,113 +13,177 @@
...
-        private final String comment;
+    @Override
+    public String toString() {
+        return asString(false);
+    }
 
-        public WithEmail(String name, String comment) {
-            this.name = name;
-            this.comment = comment;
+    public String asString(boolean ignoreEmptyValues) {
```
Comment (vanitasvitae) [inline pgpainless-core/src/main/java/org/pgpainless/key/util/UserId.java:139]: Could you add some javadoc here that documents what `ignoreEmptyValues` does?

**Review by vanitasvitae — CHANGES_REQUESTED** [review]
Nice work!
There are two small nits, but otherwise I'm quite pleased with the changes :) Thank you!

Edit: Don't worry about the travis build failure. That is caused by a bug in BC.

**Comment by IvanPizhenko** [comment]
@vanitasvitae fixed all that

**Comment by IvanPizhenko** [comment]
> Edit: Don't worry about the travis build failure. That is caused by a bug in BC.

I have locally all tests passing.

**Comment by vanitasvitae** [comment]
> I have locally all tests passing.

Yeah its happening irregularly.

**Comment by IvanPizhenko** [comment]
@vanitasvitae Thanks for merging this! Can you please make a new release?

**Comment by vanitasvitae** [comment]
I will see if I find some time this evening :)

**Pull request merged**